### PR TITLE
build: custom no catch block rule

### PR DESCRIPTION
**Pull request by sobolk**
## Problem

This PR solves two problems:
1. Adds a stricter rule to catch empty catch blocks which does not ignore comments.
2. It establishes a pattern for how to add, build and test custom rules if needed.

**Issue number, if available:**
https://github.com/aws-amplify/amplify-backend/issues/557

## Changes

1. Add a package for custom rules
2. Add rule
3. Add test
4. Integrate with linter config

**Corresponding docs PR, if applicable:**

## Validation

This PR checks exercise the rule.

Validate in IDE.
image

## Notes

When this changes goes in a full rebuild might be needed for IDE to pick up linter config

## References:

https://typescript-eslint.io/developers/custom-rules/
https://typescript-eslint.io/packages/rule-tester/#with-specific-frameworks
https://medium.com/inato/using-typescript-to-build-custom-eslint-rules-faster-53ad1c9dee2b

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: packages/eslint-rules/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "eslint-plugin-amplify-backend-rules",
+  "private": true,
+  "version": "0.0.1",
+  "type": "commonjs",
+  "main": "lib/index.js",
```
Comment (sobolk) [inline packages/eslint-rules/package.json:6]: this is needed for eslint to be able to load the plugin.

File: .eslintrc.cjs [hunk 2]
```diff
@@ -85,6 +86,7 @@ module.exports = {
     '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
     '@typescript-eslint/member-ordering': 'error',
     '@shopify/prefer-early-return': 'error',
+    'amplify-backend-rules/no-empty-catch': 'error',
```
Comment (srquinn21) [inline .eslintrc.cjs:89]: While this applies globally to all packages in the repo, have we considered making this rule on by default so no config is needed outside of import?

Comment (sobolk) [reply .eslintrc.cjs:89]: we could do something like this https://github.com/eslint-community/eslint-plugin-promise/blob/bbcfcbf85ef97cba58f4fe4b9f46c907e04bc5a6/index.js#L28-L46 
and add to this section
https://github.com/aws-amplify/amplify-backend/blob/981bd4423e4b87217cbfef51548d2f30411cc4dc/.eslintrc.cjs#L7-L13

I'll give it a try.

Comment (sobolk) [reply .eslintrc.cjs]: seems to be working, added.

**Review by srquinn21 — APPROVED** [review]

**Review by sdstolworthy — DISMISSED** [review]

**Review by sdstolworthy — APPROVED** [review]

**Review by srquinn21 — APPROVED** [review]

**Review by edwardfoyle — APPROVED** [review]
Nice!

**Pull request merged**